managed/ui/src/components/universes/ToggleUniverseState/ToggleUniverseState.js

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) YugaByte, Inc.
+// 
+// Author: Nishant Sharma(nishant.sharma@hashedin.com)
+
+import React, { Component } from 'react';
+import { YBModal, YBTextInput } from '../../common/forms/fields';
+import { getPromiseState } from '../../../utils/PromiseUtils';
+import { browserHistory } from 'react-router';
+
+class ToggleUniverseState extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      universeName: false
+    };
+  }
+
+  onChangeUniverseName = (value) => {
+    this.setState({ universeName: value });
+  };
+
+  closeDeleteModal = () => {
+    this.props.onHide();
+  };
+
+  toggleUniverseStateConfirmation = () => {
+    const {
+      universePaused,
+      universe: {
+        currentUniverse: { data }
+      }
+    } = this.props;
+    this.props.onHide();
+    universePaused
+      ? this.props.submitRestartUniverse(data.universeUUID)
+      : this.props.submitPauseUniverse(data.universeUUID);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      (getPromiseState(prevProps.universe.pauseUniverse).isLoading() &&
+        getPromiseState(this.props.universe.pauseUniverse).isSuccess()) ||
+      (getPromiseState(prevProps.universe.restartUniverse).isLoading() &&
+        getPromiseState(this.props.universe.restartUniverse).isSuccess())
+    ) {
+      this.props.fetchUniverseMetadata();
+      browserHistory.push('/universes');
+    }
+  }
```

Comment:
1. I took an example of the working component which is 'Delete'. And it will redirect only if the condition satisfied. 

2. I didn't get the 'passing onSubmit function that gets run if the submit is successful`. Can you elaborate on this and please tell me if there's any working example of this to follow.